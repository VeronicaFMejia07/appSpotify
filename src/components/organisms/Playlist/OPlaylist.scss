//--MOBILE 320PX--
.section__MPlaylist {
    display: flex;
    width: 100%;
    min-height: 100vh;
    background-color: #ffb271;
    padding-bottom: 3rem;
    padding-top: 1rem;
}

.section__playlist {
    display: grid;
    width: 100%;
    grid-template-columns: 50% 50%;
    background-color: #ffb271;
    margin-bottom: 1rem;
}

//--TABLET 768PX--
@media (min-width: 768px) {
    .section__MPlaylist{
        width: 100%;
        display: flex;
        justify-content: flex-end;
    }
    .section__playlist {
        width: 80%;
        grid-template-columns: 33% 33% 33%;
    }
}

//--LAPTOP 1024PX--
@media (min-width: 1024px) {
    .section__playlist {
        width: 85%; 
        grid-template-columns: 33% 33% 33%;
    }
}

//--DESKTOP 1366PX--
@media (min-width: 1366px){
    .section__MPlaylist{
        width: 100%;
    }
    .section__playlist {
        width: 85%; 
        grid-template-columns: 20% 20% 20% 20% 20%;
    }
}