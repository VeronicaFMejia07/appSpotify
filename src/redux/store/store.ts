import { configureStore } from "@reduxjs/toolkit";
import {userSlice} from "redux/slices/userSlice";
import {playlistSlice } from "redux/slices/playlistSlice";
import {favoritesSlice} from "redux/slices/favoritesSlice";

export const store= configureStore({
    reducer:{
        playlist: playlistSlice.reducer,
        favorites: favoritesSlice.reducer,
        user: userSlice.reducer,
    }
})

export type RootState = ReturnType<typeof store.getState>
export type AppDispatch = typeof store.dispatch
export default store