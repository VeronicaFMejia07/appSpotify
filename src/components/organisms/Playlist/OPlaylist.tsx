import React, { FC, useEffect, useState } from 'react'
import MPlaylist from 'components/molecules/PlayList/MPlaylist';
import 'components/organisms/Playlist/OPlaylist.scss';
import { selectPlaylistInfo} from 'redux/slices/playlistSlice';
import {useAppSelector } from 'redux/hooks/hooks';

//FunciÃ³n que recibe la lista de canciones y las recorre por medio del map
const OPlaylist= () => {
  const { playlist } = useAppSelector(selectPlaylistInfo);//TRAE LA PLAYLIST-SELECTOR

  return (
    <section className='section__MPlaylist'>
      <article className='section__playlist'>
        {
          playlist.map((item, index) => (
            <MPlaylist key={index} item={item} />
          ))
        }
      </article>
    </section>
  )
}

export default OPlaylist