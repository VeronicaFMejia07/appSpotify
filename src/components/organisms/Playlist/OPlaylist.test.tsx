import React from 'react';
import { render } from '@testing-library/react';
import OPlaylist from 'components/organisms/Playlist/OPlaylist';
import { Provider } from 'react-redux';
import store from 'redux/store/store';

//Describir la prueba que se este haciendo
describe('Pruebas en Oplaylist', () => {
  test('Renderiza componente--OPlaylist ', () => {
    const { container } = render(
      <Provider store={store}>
        <OPlaylist />
      </Provider>
    )

    //Evalua que el container sea igual a lo que contiene el componente
    expect(container).toMatchSnapshot();
  })

})


