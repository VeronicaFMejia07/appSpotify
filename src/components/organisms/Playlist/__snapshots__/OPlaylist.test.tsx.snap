// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en Oplaylist Renderiza componente--OPlaylist  1`] = `
<div>
  <section
    class="section__MPlaylist"
  >
    <article
      class="section__playlist"
    />
  </section>
</div>
`;
